libparsec/crates/client/src/certif/device.rs

```diff
@@ -0,0 +1,161 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::sync::Arc;
+
+use libparsec_client_connection::{AuthenticatedCmds, ConnectionError};
+use libparsec_platform_storage::certificates::GetCertificateError;
+use libparsec_protocol::authenticated_cmds::v4::device_create;
+use libparsec_types::{
+    anyhow, thiserror, Bytes, CertificateSignerOwned, DateTime, DeviceCertificate, LocalDevice,
+    MaybeRedacted, SigningKeyAlgorithm, UserID,
+};
+
+use super::{
+    encrypt::CertifEncryptForUserError, greater_timestamp, CertificateBasedActionOutcome,
+    GreaterTimestampOffset, InvalidCertificateError,
+};
+
+pub async fn create_new_device(
+    cmds: Arc<AuthenticatedCmds>,
+    new_device: LocalDevice,
+    author: Arc<LocalDevice>,
+) -> Result<CertificateBasedActionOutcome, CertifDeviceError> {
```

Comment:
Maybe rename this function `register_new_device`, since it receives a `new_device` as an argument (I understand that it's going to be sending a `create_device` request to the server, but in this context it's unclear whether `create` refers to the local device creation or its registration on the server)